checks/tools/io/cdo.py

```diff
@@ -21,60 +20,100 @@
             self.valid_prog_environs = ['cpeGNU']
-            self.modules = ['CDO']
+        elif self.current_system.name in ['manali', 'balfrin']:
+            self.valid_prog_environs = ['PrgEnv-gnu']
+            self.squashfs_script = './exe.sh'
+            # until VCMSA-102 is resolved:
+            if self.current_system.name in ['balfrin']:
+                self.prerun_cmds += [
+                    r'echo "export LD_LIBRARY_PATH=\$LD_LIBRARY_PATH:'
+                    rf'\$SCRATCH/balfrin/lib64" >> {self.squashfs_script}'
+                ]
+            self.prerun_cmds += [
+                f'grep ^module rfm_job.sh |grep -v unuse >> '
+                f'{self.squashfs_script}',
+                f'echo "\\$*" >> {self.squashfs_script}',
+            ]
         else:
             self.valid_prog_environs = ['builtin']
-            self.modules = ['CDO']
-
-        self.maintainers = ['SO', 'CB']
-        self.tags = {'production', 'mch', 'external-resources'}
 
 
 # Check that the netCDF loaded by the CDO module supports the nc4 filetype
 # (nc4 support must be explicitly activated when the netCDF library is
 # compiled...).
 @rfm.simple_test
 class CDO_DependencyTest(CDOBaseTest):
-    def __init__(self):
-        super().__init__()
-        self.descr = ('verifies that the netCDF loaded by the CDO module '
-                      'supports the nc4 filetype')
-        self.sourcesdir = None
-        self.executable = 'nc-config'
-        self.executable_opts = ['--has-nc4']
-        self.sanity_patterns = sn.assert_found(r'^yes', self.stdout)
+    descr = ('verifies that the netCDF loaded by the CDO module '
+             'supports the nc4 filetype')
+    sourcesdir = None
+
+    @run_before('run')
+    def setup_per_partition_name(self):
+        if 'squashfs' in self.current_partition.name:
+            self.executable = f'bash {self.squashfs_script}'
+            self.executable_opts = ['nf-config', '--has-nc4']
+        else:
+            self.executable = 'nc-config'
+            self.executable_opts = ['--has-nc4']
+
+    @sanity_function
+    def assert_output(self):
+        return sn.assert_found(r'^yes', self.stdout)
 
 
 @rfm.simple_test
 class CDO_NC4SupportTest(CDOBaseTest):
-    def __init__(self):
-        super().__init__()
-        self.descr = ('verifies that the CDO supports the nc4 filetype')
-        self.sourcesdir = None
-        self.executable = 'cdo'
-        self.executable_opts = ['-V']
-        self.sanity_patterns = sn.assert_found(r'^Filetypes:.*\snc4\s'
-                                               r'nc4c\W', self.stderr)
+    descr = ('verifies that the CDO supports the nc4 filetype')
+    sourcesdir = None
+
+    @run_before('run')
+    def setup_per_partition_name(self):
+        if 'squashfs' in self.current_partition.name:
+            self.executable = f'bash {self.squashfs_script}'
+            self.executable_opts = ['cdo', '-V', '2>&1']
```

Comment:
Why are you redirecting the output here? Can't we just read the stderr?

Reply:
older version of cdo (on tsa) use stderr but latest versions use stdout hence it's better to force it.